sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -259,4 +264,50 @@ export class InstitutionController extends BaseController {
       body.isActive,
     );
   }
+
+  @Get("/my-details")
+  async getMyDetails(
+    @UserToken() token: IInstitutionUserToken,
+  ): Promise<InstitutionUserAndAuthDetails> {
+    // Get logged in user and location details with auth
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(
+        token.userName,
+      );
+    const user = {
+      user: {
+        firstName: institutionUser.user?.firstName,
+        lastName: institutionUser.user?.lastName,
+        userName: token.userName ?? institutionUser.user?.userName,
+        isActive: institutionUser.user?.isActive,
+        isAdmin: token.authorizations.isAdmin,
+        id: institutionUser.user?.id,
```

Comment:
Same from the previous comment, I believe that we should not expose this outside the API.